### PR TITLE
CB-29-Add-comments

### PR DESCRIPTION
**Pull request by banaskrzysztof**
https://team-1626248039387.atlassian.net/browse/CB-29?atlOrigin=eyJpIjoiNWQ4ZjMzNjYyNDhlNGExMzhhZDY0NDk4NGM2NGU2MWYiLCJwIjoiaiJ9

### CODE REVIEW

File: src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      author: new FormControl(''),
+      body: new FormControl(''),
+    })
+  }
+
+  ngOnInit(): void {
+    this.getAllComments();
+  }
+
+  getAllComments() {
```
Comment (KrzysiekLipiec) [inline src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts:48]: add return value

File: src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    this.commentsService.getComments().subscribe(result => {
+      this.comments = result;
+    });
+  }
+
+  deleteComment(id: number) {
```
Comment (KrzysiekLipiec) [inline src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts:54]: add return value

File: src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  editComment(id: number): void {
+    this.showEdit();
+    this.commentsService.getComment(id).subscribe(result => {
+      this._comment = result;
+      this._comment.updated_at = new Date();
+    })
+  }
+
+  onSubmit() {
```
Comment (KrzysiekLipiec) [inline src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts:70]: add return value

**Review by KrzysiekLipiec — APPROVED** [review]

File: src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  constructor(private fb: FormBuilder,
+              private commentsService: RecipeDetailCommentsService) {
+    this.commentForm = this.fb.group({
+      author: new FormControl(''),
+      body: new FormControl('')
+    })
+  }
+
+  onSubmit() {
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts:37]: return type

File: src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      this.commentsService.addComment(this.comment).subscribe();
+  }
+
+  onBack() {
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts:41]: return type

File: src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+@Component({
+  selector: 'app-recipe-detail-add-comment',
+  templateUrl: './recipe-detail-add-comment.component.html',
+  styleUrls: ['./recipe-detail-add-comment.component.scss']
+})
+export class RecipeDetailAddCommentComponent {
+  commentForm: FormGroup;
+
+  @Input() sComments: number | undefined;
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts:14]: unused?

File: src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  templateUrl: './recipe-detail-comments-list.component.html',
+  styleUrls: ['./recipe-detail-comments-list.component.scss']
+})
+
+export class RecipeDetailCommentsListComponent implements OnInit {
+  comments: Comment[] = [];
+  show: boolean = true;
+  commentForm: FormGroup;
+
+  @Input() sComments: number | undefined;
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts:17]: unused?

File: src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  set comment(comment: Comment) {
+    this._comment = comment;
+  }
+
+  private _comment: Comment = {
+    id: 0,
+    author: "",
+    body: "",
+    create_at: new Date(),
+    updated_at: null,
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-add-comment/recipe-detail-add-comment.component.ts:26]: why do you need a setter and getter?

File: src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    this._comment = comment;
+  }
+
+  private _comment: Comment = {
+    id: 0,
+    author: "",
+    body: "",
+    create_at: null,
+    updated_at: null,
+  }
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-comments-list/recipe-detail-comments-list.component.ts:34]: why do you need setter and getter

**Pull request merged**